src/org/openbravo/erpCommon/ad_forms/AcctServer.java

```
@@ -2353,93 +2367,114 @@ public void setMessageResult(ConnectionProvider conn, String _strStatus, String
+      case STATUS_NotCalculatedCost:
+        return (params.isEmpty()) ? "@NotCalculatedCost@" : "@NotCalculatedCostWithTransaction@";
+      case STATUS_InvalidCost:
+        if (params.isEmpty()) {
+          return "@InvalidCost@";
+        } else {
+          // Translate account name from messages
+          params.put(ACCOUNT_PARAM,
+              Utility.parseTranslation(conn, vars, vars.getLanguage(), params.get(ACCOUNT_PARAM)));
+          return "@InvalidCostWhichProduct@";
+        }
+      case STATUS_NoRelatedPO:
+        return params.isEmpty() ? "@GoodsReceiptTransactionWithNoPO@" : "@GoodsReceiptTransactionWithNoPOWichProduct@";
+      case STATUS_DocumentDisabled:
+        messageResult.setType(WARNING);
+        return "@DocumentDisabled@";
+      case STATUS_BackgroundDisabled:
+        messageResult.setType(WARNING);
+        return "@BackgroundDisabled@";
+      case STATUS_InvalidAccount:
+        if (params.isEmpty()) {
+          return "@InvalidAccount@";
+        } else {
+          // Translate account name from messages
+          params.put(ACCOUNT_PARAM,
+              Utility.parseTranslation(conn, vars, vars.getLanguage(), params.get(ACCOUNT_PARAM)));
+          return "@InvalidWhichAccount@";
+        }
+      case STATUS_NODOCTYPE:
+        return "@NoDocTypeForDocument@";
+      case STATUS_PeriodClosed:
+        return "@PeriodNotAvailable@";
+      case STATUS_NotConvertible:
+        return NOT_CONVERTIBLE_MESSAGE;
+      case STATUS_NotBalanced:
+        return "@NotBalanced@";
+      case STATUS_NotPosted:
+        return "@NotPosted@";
+      case STATUS_PostPrepared:
+        return "@PostPrepared@";
+      case STATUS_Posted:
+        return "@Posted@";
+      case STATUS_TableDisabled:
+        params.put("Table", tableName);
+        messageResult.setType(WARNING);
+        return "@TableDisabled@";
+      case STATUS_NoAccountingDate:
+        return "@NoAccountingDate@";
+      default:
+        return "";
```

This case should return an error as this case would  only be obtained if the status is not in the list